distribution/lib/Standard/Visualization/0.0.0-dev/src/Scatter_Plot.enso

```diff
@@ -66,13 +69,25 @@ type Point_Data
     ## PRIVATE
     fallback_column : Table -> Column ! No_Fallback_Column
     fallback_column self table = case self of
-        Point_Data.X -> Point_Data.iota table.row_count
+        Point_Data.X ->
+            candidate = table.columns.first
+
+            if candidate.is_numeric then candidate else Point_Data.iota table.row_count
         Point_Data.Y ->
             x_column = Point_Data.X.lookup_in table
             candidates = table.columns
             is_good_enough c = c.is_numeric && c.name != x_column.name
             is_good c = is_good_enough c && (self.is_recognized c).not
 
+            candidates.find if_missing=(Error.throw Nothing) is_good . catch_ <| candidates.find is_good_enough
+        Point_Data.Y_multi _ ->
+            x_column = Point_Data.X.lookup_in table
+            y_column = Point_Data.Y.lookup_in table
+            candidates_filtered = table.columns.filter (c-> c.is_numeric && c.name != x_column.name && c.name != y_column.name)
+            candidates = candidates_filtered.slice self.number (candidates_filtered.length)
+            is_good_enough c = c.is_numeric && c.name != x_column.name && c.name != y_column.name
+            is_good c = is_good_enough c && (self.is_recognized c).not
+
             candidates.find if_missing=(Error.throw Nothing) is_good . catch_ <| candidates.find is_good_enough
```

Comment:
Hmm I just realized - `candidates` is a slice of `candidates_filtered`.

All elements of `candidates_filtered` satisfy the `is_good_enough` predicate by definition.

So this can be further simplified:
- `is_good` does not need to check for `is_good_enough` - all elements on which we run `is_good` must already satisfy ``is_good_enough c == True`.
- the last line could become:
```suggestion
            candidates.find if_missing=candidates.first is_good
```

Because any value on `candidates` will satisfy `is_good_enough` so the second fallback `find` we had, would actually always return the first value.